### ISSUE TITLE
False "react/prop-types [PROP_NAME] is missing in props validation" in a class based component having methods with destructured annotated params (Flow)

### ISSUE THREAD

**Issue by jtraub** [yannickcr/eslint-plugin-react#729]
Consider a following snippet

```
import React, { Component, PropTypes } from 'react';

type Target = { target: EventTarget }

class MyComponent extends Component {
  static propTypes = {
    children: PropTypes.any,
  }

  handler({ target }: Target) {
    if (target) {
      // do something
    }
  }

  render() {
    return (
      <div>
        {this.props.children}
      </div>
    );
  }

}

export default MyComponent;

```

Linting gives false error

```
./node_modules/.bin/eslint src/components/Test.js 

/mount/remote/test_env/src/components/Test.js
  19:21  error  'children' is missing in props validation  react/prop-types

✖ 1 problem (1 error, 0 warnings)
```

A quick look up shows that the problem is caused by [this commit](https://github.com/yannickcr/eslint-plugin-react/commit/259fab4ed1cb6299eb7059b90c995684a0a4baed) which fixes a similar problem.

P.S. This report is based on [StackOverflow question](http://stackoverflow.com/q/38684925/471899).
